autodiff.py

```diff
@@ -811,14 +834,20 @@ def set_inputs(self, nodes):
         self.inputs = nodes
         node_names = [node.name for node in nodes]
         self.name = EinsumNode._name_generator(self.einsum_subscripts,
-                                               node_names)
+                                               node_names, self.format)
         self.shape = self._output_shape(self.einsum_subscripts, nodes)
 
     def compute(self, input_vals):
         """Given values of input nodes, return result of matrix multiplication."""
         for val in input_vals:
             assert T.is_tensor(val)
-        return T.einsum(self.einsum_subscripts, *input_vals)
+        if self.format == "dense":
+            return T.einsum(self.einsum_subscripts, *input_vals)
+        else:
+            assert T.support_sparse_format
+            return T.einsum(self.einsum_subscripts,
+                            *input_vals,
+                            out_format=self.format)
```

Comment:
What about just throwing errors here and add a TODO(): Call TACO API.

Reply:
I have removed the previous sparse einsum implementation for numpy, it now works for taco.